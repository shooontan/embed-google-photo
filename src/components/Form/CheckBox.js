// @flow
import * as React from 'react';

type Props = {
  label?: string,
};

const CheckBox = (props: Props) => {
  const { label } = props;
  return (
    <p>
      <input type="checkbox" {...props} />
      {label}
    </p>
  );
};

CheckBox.defaultProps = {
  label: '',
};

export default CheckBox;
