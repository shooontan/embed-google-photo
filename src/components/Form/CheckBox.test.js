// @flow
import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import CheckBox from './CheckBox';

Enzyme.configure({ adapter: new Adapter() });

test('CheckBox Component', () => {
  let testRenderer = TestRenderer.create(<CheckBox />);
  expect(testRenderer).toMatchSnapshot();

  testRenderer = TestRenderer.create(<CheckBox label="checkbox" />);
  expect(testRenderer).toMatchSnapshot();

  const mock = jest.fn();
  testRenderer = shallow(<CheckBox onChange={mock} />);
  testRenderer.find('input').simulate('change');
  expect(mock).toHaveBeenCalled();
});
