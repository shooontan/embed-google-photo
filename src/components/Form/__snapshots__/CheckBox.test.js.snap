// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component 1`] = `
<p>
  <input
    label=""
    type="checkbox"
  />
  
</p>
`;

exports[`CheckBox Component 2`] = `
<p>
  <input
    label="checkbox"
    type="checkbox"
  />
  checkbox
</p>
`;
